### PR TITLE
Opt executor test fixes, code cleanup

### PR DESCRIPTION
**Pull request by mzabaluev**
# Summary
- **Categories**: `protocol-units`.

In `test_pipe_mempool_with_malformed_transaction`, check the transaction submission status that the transaction is pre-validated.
This makes the test fail rather than hang up.

Also commit some drive-by code cleanups.

# Testing

`cargo check -p maptos-opt-executor`

# Outstanding issues

The test is still broken, we need a setup that passes the validator with a randomly generated transaction.

### CODE REVIEW

File: protocol-units/execution/opt-executor/src/executor/initialization.rs [hunk 1]
```diff
@@ -167,7 +167,7 @@ impl Executor {
 		// replace the db path with a temporary directory
 		let value = tempfile::tempdir()?.into_path(); // todo: this works because it's at the top level, but won't be cleaned up automatically
 		maptos_config.chain.maptos_db_path.replace(value);
+		dbg!(&maptos_config.chain.maptos_db_path);
```
Comment (l-monninger) [inline protocol-units/execution/opt-executor/src/executor/initialization.rs:170]: Remember to remove this, unless you have a good reason to keep it.

**Review by l-monninger — CHANGES_REQUESTED** [review]

File: protocol-units/execution/opt-executor/src/executor/initialization.rs (hunk 1 above)
Comment (mzabaluev) [reply protocol-units/execution/opt-executor/src/executor/initialization.rs]: Done in the subsequent commit.

File: protocol-units/execution/opt-executor/src/executor/services.rs [hunk 2]
```diff
@@ -35,49 +35,43 @@ impl Executor {
...
+		let (mut executor, _tempdir) = Executor::try_test_default(GENESIS_KEYPAIR.0.clone())?;
 		let server_executor = executor.clone();
 
 		let handle = tokio::spawn(async move {
 			server_executor.run_service().await?;
 			Ok(()) as Result<(), anyhow::Error>
 		});
 
-		let user_transaction = create_signed_transaction(0, executor.maptos_config.chain.maptos_chain_id.clone());
+		let user_transaction = create_signed_transaction(1, &executor.maptos_config);
```
Comment (l-monninger) [inline protocol-units/execution/opt-executor/src/executor/services.rs:74]: Do we actually need `1` here?

**Review by l-monninger — APPROVED** [review]
LGTM

File: protocol-units/execution/opt-executor/src/executor/services.rs (hunk 2 above)
Comment (mzabaluev) [reply protocol-units/execution/opt-executor/src/executor/services.rs:74]: No, 0 works just as well. I have changed this helper function for consistency with the one in the sibling module `transaction_pipe`; perhaps a shared test util module would be better, but we may need different transaction data for different tests, too.

Comment (l-monninger) [reply protocol-units/execution/opt-executor/src/executor/services.rs]: Shared is better ofc. But, don't worry about it. Just wanted to clarify conceptually.

**Pull request merged**